src/Microsoft.ML.Core/Data/IEstimator.cs

```diff
@@ -45,6 +45,12 @@ public Column(string name, VectorKind vecKind, DataKind itemKind, bool isKey, st
                 IsKey = isKey;
                 MetadataKinds = metadataKinds;
             }
+
+            public Column CloneWithNewName(string newName)
```

Comment:
>CloneWithNewName [](start = 26, length = 16)

maybe this calls for mutable Column object? 
If we don't do this, I guess you just have to write line 52 everywhere :)